beacon-chain/core/state/transition_test.go

```diff
@@ -479,3 +479,137 @@ func TestProcessBlock_PassesProcessingConditions(t *testing.T) {
 		t.Errorf("Expected block to pass processing conditions: %v", err)
 	}
 }
+
+func TestProcessEpoch_PassesProcessingConditions(t *testing.T) {
```

Comment:
Only two tests? It seems like there are a lot of error branches to cover in the state transition func. However, the smaller components are already thoroughly tested. Despite this, it would be cool to see a progression over the error branches all the way up to the PassesProcessingConditions test. Can we try this out the same way it was done for block processing? Or we can also do it in a separate PR. Doing that would also maintain the coverage near 100%.

Reply:
I agree more tests would be helpful. Although everything gets called within  `NewEpochTransition` individually has >95% test coverage. Problem is it is tough to test every  error returns within `NewEpochTransition`, the reason being:

```go
func NewEpochTransition() error {

if err := processXXX, err != nil { 
   return err
}

if err := processYYY, err != nil { 
   return err
}

if err := processZZZ, err != nil { 
   return err
}
}
```

`processXXX`'s error condition (e.g. can't get proposer index)  is the same as `processYYY` and  `processZZZ`, means we will never hit `processYYY` and `processZZZ` error conditions